src/core/mac/mac_frame.cpp

```diff
@@ -1061,49 +1061,37 @@ const TimeIe *Frame::GetTimeIe(void) const
 #if OPENTHREAD_CONFIG_MULTI_RADIO
 uint16_t Frame::GetMtu(void) const
 {
-    uint16_t mtu;
+    uint16_t mtu = OT_RADIO_FRAME_MAX_SIZE;
 
     switch (GetRadioType())
     {
-#if OPENTHREAD_CONFIG_RADIO_LINK_IEEE_802_15_4_ENABLE
-    case kRadioTypeIeee802154:
-        mtu = OT_RADIO_FRAME_MAX_SIZE;
-        break;
-#endif
-
 #if OPENTHREAD_CONFIG_RADIO_LINK_TREL_ENABLE
     case kRadioTypeTrel:
         mtu = Trel::Link::kMtuSize;
         break;
 #endif
 
     default:
```

Comment:
Since we are removing the `OT_ASSERT`, I suggest we also remove the `default` case fully. 

This way, if a valid `enum` case is not included in `switch`, we will get an warning (on some toolchain).

Reply:
@simonlingoogle , thoughts on removing the default case?

Reply:
To remove default, we need to add `case kRadioTypeIeee802154` back.

Reply:
Verified by https://github.com/openthread/ot-br-posix/pull/803/checks that the current change compiles without issue.